public/app/plugins/datasource/prometheus/datasource.ts

```diff
@@ -340,9 +344,9 @@ export class PrometheusDatasource extends DataSourceApi<PromQuery, PromOptions>
     let interval = kbn.interval_to_seconds(options.interval);
     // Minimum interval ("Min step"), if specified for the query or datasource. or same as interval otherwise
     const minInterval = kbn.interval_to_seconds(
-      this.templateSrv.replace(target.interval, options.scopedVars) || options.interval
+      templateSrv.replace(target.interval, options.scopedVars) || options.interval
     );
-    const intervalFactor = target.intervalFactor || 1;
+    const intervalFactor = target.intervalFactor ?? 1;
```

Comment:
sure we can replace it here? Is `target.intervalFactor == 0` a valid value?

Reply:
Good catch, I'll change